### PR TITLE
docs: LSIF on GitHub with sourcegraph.com endpoint and TOS

### PR DESCRIPTION
**Pull request by rvantonder**
- Explain availability of sourcegraph.com as an LSIF upload endpoint
- Explain how we treat data and link to TOS
- Point out email for support and/or if data needs to be deleted

### CODE REVIEW

File: doc/user/code_intelligence/lsif_on_github.md [hunk 1]
```diff
@@ -32,11 +32,17 @@ jobs:
...
 
 ![img/workflow.png](img/workflow.png)
 
 After the workflow succeeds, you should see LSIF-powered code intelligence on your repository on Sourcegraph.com or on GitHub with the [Sourcegraph browser extension](../../integration/browser_extension.md).
+
+## Using an upload endpoint
+
+LSIF data can be uploaded to a self-deployed Sourcegraph instance or to [sourcegraph.com](https://sourcegraph.com). Using the [sourcegraph.com](https://sourcegraph.com) endpoint will surface code intelligence for your public repositories directly on GitHub via the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension) and at `https://sourcegraph.com/github.com/<your-username>/<your-repo>`. 
+
+Using the [sourcegraph.com](https://sourcegraph.com) endpoint is free and your LSIF data is treated as User-Generated Content (you own it, as covered our [sourcegraph.com terms of service](https://about.sourcegraph.com/terms-dotcom#3-proprietary-rights-and-licenses). If you run into trouble, or a situation arises where you need all of your LSIF data expunged, please reach out to us at [support@sourcegraph.com](mailto:support@sourcegraph.com).
```
Comment (rvantonder) [inline doc/user/code_intelligence/lsif_on_github.md]:
```suggestion
Using the [sourcegraph.com](https://sourcegraph.com) endpoint is free and your LSIF data is treated as User-Generated Content (you own it, as covered our [sourcegraph.com terms of service](https://about.sourcegraph.com/terms-dotcom#3-proprietary-rights-and-licenses)). If you run into trouble, or a situation arises where you need all of your LSIF data expunged, please reach out to us at [support@sourcegraph.com](mailto:support@sourcegraph.com).
```

File: doc/user/code_intelligence/lsif_on_github.md [hunk 2]
```diff
@@ -1,6 +1,6 @@
 # LSIF on GitHub
 
-You can use [GitHub Actions](https://help.github.com/en/github/automating-your-workflow-with-github-actions/about-github-actions) to (1) generate LSIF data and (2) upload it to your Sourcegraph instance.
+You can use [GitHub Actions](https://help.github.com/en/github/automating-your-workflow-with-github-actions/about-github-actions) to (1) generate LSIF data and (2) upload it to a [Sourcegraph endpoint](#using-an-upload-endpoint).
```
Comment (creachadair) [inline doc/user/code_intelligence/lsif_on_github.md]: Thanks for adding the additional context. 👍

**Comment by rvantonder** [comment]
@dadlerj can you take a quick look 🙏 After this is in I feel good about making PRs to other projects.

File: doc/user/code_intelligence/lsif_on_github.md [hunk 3]
```diff
@@ -32,11 +32,17 @@ jobs:
...
 
 Once that workflow is committed to your repository, you will start to see LSIF workflows in the Actions tab of your repository (e.g. https://github.com/sourcegraph/sourcegraph/actions).
 
 ![img/workflow.png](img/workflow.png)
 
 After the workflow succeeds, you should see LSIF-powered code intelligence on your repository on Sourcegraph.com or on GitHub with the [Sourcegraph browser extension](../../integration/browser_extension.md).
+
+## Using an upload endpoint
+
+LSIF data can be uploaded to a self-deployed Sourcegraph instance or to [sourcegraph.com](https://sourcegraph.com). Using the [sourcegraph.com](https://sourcegraph.com) endpoint will surface code intelligence for your public repositories directly on GitHub via the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension) and at `https://sourcegraph.com/github.com/<your-username>/<your-repo>`.
```
Comment (dadlerj) [inline doc/user/code_intelligence/lsif_on_github.md]:
```suggestion
LSIF data can be uploaded to a self-hosted Sourcegraph instance or to [Sourcegraph.com](https://sourcegraph.com). Using the [Sourcegraph.com](https://sourcegraph.com) endpoint will surface code intelligence for your public repositories directly on GitHub via the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension) and at `https://sourcegraph.com/github.com/<your-username>/<your-repo>`. 
```

Changes just based on style guide for referring to our products: https://about.sourcegraph.com/handbook/communication/style_guide#referring-to-the-product-and-features

File: doc/user/code_intelligence/lsif_on_github.md [hunk 4]
```diff
@@ -32,11 +32,17 @@ jobs:
...
 
 ![img/workflow.png](img/workflow.png)
 
 After the workflow succeeds, you should see LSIF-powered code intelligence on your repository on Sourcegraph.com or on GitHub with the [Sourcegraph browser extension](../../integration/browser_extension.md).
+
+## Using an upload endpoint
+
+LSIF data can be uploaded to a self-deployed Sourcegraph instance or to [sourcegraph.com](https://sourcegraph.com). Using the [sourcegraph.com](https://sourcegraph.com) endpoint will surface code intelligence for your public repositories directly on GitHub via the [Sourcegraph browser extension](https://docs.sourcegraph.com/integration/browser_extension) and at `https://sourcegraph.com/github.com/<your-username>/<your-repo>`. 
+
+Using the [sourcegraph.com](https://sourcegraph.com) endpoint is free and your LSIF data is treated as User-Generated Content (you own it, as covered our [sourcegraph.com terms of service](https://about.sourcegraph.com/terms-dotcom#3-proprietary-rights-and-licenses)). If you run into trouble, or a situation arises where you need all of your LSIF data expunged, please reach out to us at [support@sourcegraph.com](mailto:support@sourcegraph.com).
```
Comment (dadlerj) [inline doc/user/code_intelligence/lsif_on_github.md]:
```suggestion
Using the [Sourcegraph.com](https://sourcegraph.com) endpoint is free and your LSIF data is treated as User-Generated Content (you own it, as covered in our [Sourcegraph.com terms of service](https://about.sourcegraph.com/terms-dotcom#3-proprietary-rights-and-licenses)). If you run into trouble, or a situation arises where you need all of your LSIF data expunged, please reach out to us at [support@sourcegraph.com](mailto:support@sourcegraph.com).
```

**Pull request merged**